src/main/java/com/google/googleidentity/oauth2/exception/OAuth2ExceptionHandler.java

```diff
@@ -0,0 +1,174 @@
+/*
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package com.google.googleidentity.oauth2.exception;
+
+import com.google.googleidentity.oauth2.util.OAuth2ParameterNames;
+import net.minidev.json.JSONObject;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.utils.URIBuilder;
+
+import java.net.URISyntaxException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * class to handle {@link OAuth2Exception}
+ */
+public final class OAuth2ExceptionHandler {
+
+    private static final Logger log = Logger.getLogger("OAuth2ExceptionHandler");
+
+    private static final String INVALID_REQUEST = "invalid_request";
+    private static final String INVALID_GRANT = "invalid_grant";
+    private static final String UNAUTHORIZED_CLIENT = "unauthorized_client";
+    private static final String INVALID_CLIENT = "invalid_client";
+    private static final String UNSUPPORTED_RESPONSE_TYPE = "unsupported_response_type";
+    private static final String INVALID_SCOPE = "invalid_scope";
+    private static final String ACCESS_DENIED = "access_denied";
+    private static final String UNSUPPORTED_GRANT_TYPE = "unsupported_grant_type";
+
+    public enum ErrorCode{
+        NO_RESPONSE_TYPE,
+        UNSUPPORTED_RESPONSE_TYPE,
+        NO_CLIENT_ID,
+        NONEXISTENT_CLIENT_ID,
+        UNAUTHORIZED_CLIENT,
+        NO_REDIRECT_URI,
+        REDIRECT_URI_MISMATCH,
+        INVALID_SCOPE,
+        ACCESS_DENIED,
+        NO_GRANT_TYPE,
+        UNSUPPORTED_GRANT_TYPE,
+        NONEXISTENT_CODE,
+        CODE_CLIENT_MISMATCH,
+        CODE_REDIRECT_URI_MISMATCH,
+        INVALID_CLIENT,
+        NONEXISTENT_REFRESH_TOKEN,
+        REFRESH_TOKEN_CLIENT_MISMATCH
+    }
+
+    public static int getHttpCode(ErrorCode code){
+        if(code.equals(ErrorCode.INVALID_CLIENT)){
+            return HttpStatus.SC_UNAUTHORIZED;
+        }
+        else{
+            return HttpStatus.SC_BAD_REQUEST;
+        }
+    }
+
+    public static String getErrorType(ErrorCode code){
+        switch(code){
+            case NO_RESPONSE_TYPE:
+            case NO_CLIENT_ID:
+            case NONEXISTENT_CLIENT_ID:
+            case NO_REDIRECT_URI:
+            case REDIRECT_URI_MISMATCH: return INVALID_REQUEST;
```

Comment:
statement should be in new line
```suggestion
            case REDIRECT_URI_MISMATCH: 
              return INVALID_REQUEST;
```
Same for others